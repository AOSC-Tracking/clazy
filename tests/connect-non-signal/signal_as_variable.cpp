#include <QtCore/QObject>
#include <QtCore/QTimer>




int main()
{
    auto *o = new QObject();
    auto d = &QObject::destroyed;
    QObject::connect(o, d, []() {}); // OK

    auto p = &QObject::property;
    QObject::connect(o, p, []() {}); // WARN, we use a getter here

    auto overload = qOverload<>(&QTimer::start); // Getting the method decl does not work in Qt5 for some reason
    QObject::connect(new QTimer, overload, []() {}); // WARN

    auto overload_static_cast = static_cast<void (QTimer::*)()>(&QTimer::start);
    QObject::connect(new QTimer, overload_static_cast, []() {}); // WARN

    auto overload_ok = qOverload<QObject*>(&QObject::destroyed);
    QObject::connect(o, overload_ok, []() {}); // WARN
}
