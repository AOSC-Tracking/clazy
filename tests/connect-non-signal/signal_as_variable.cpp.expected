connect-non-signal/signal_as_variable.cpp:14:5: warning: QObject::property is not a signal [-Wclazy-connect-non-signal]
connect-non-signal/signal_as_variable.cpp:17:5: warning: QTimer::start is not a signal [-Wclazy-connect-non-signal]
connect-non-signal/signal_as_variable.cpp:20:5: warning: QTimer::start is not a signal [-Wclazy-connect-non-signal]
